server/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java

```diff
@@ -266,6 +266,16 @@ public static String threadName(final String nodeName, final String namePrefix)
         return "elasticsearch" + (nodeName.isEmpty() ? "" : "[") + nodeName + (nodeName.isEmpty() ? "" : "]") + "[" + namePrefix + "]";
     }
 
+    public static String executorName(String threadName) {
```

Comment:
I suggest to also copy the test over to this PR, I think this will go first.

Reply:
Good point :+1: